runtime/src/main/java/org/corfudb/runtime/collections/DiskBackedCorfuTable.java

```diff
@@ -0,0 +1,491 @@
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static org.corfudb.runtime.CorfuOptions.ConsistencyModel.READ_COMMITTED;
+
+@Slf4j
+@AllArgsConstructor
+@Builder(toBuilder = true)
+public class DiskBackedCorfuTable<K, V> implements
+        SnapshotGenerator<DiskBackedCorfuTable<K, V>>,
+        ViewGenerator<DiskBackedCorfuTable<K, V>>,
+        ConsistencyView {
+
+    private static final HashFunction murmurHash3 = Hashing.murmur3_32();
+    private static final String DISK_BACKED = "diskBacked";
+    private static final String TRUE = "true";
+    private static final int BOUND = 100;
+    private static final int SAMPLING_RATE = 40;
+
+    static {
+        RocksDB.loadLibrary();
+    }
+
+    private final WriteOptions writeOptions = new WriteOptions()
+            .setDisableWAL(true)
+            .setSync(false);
+    private final RocksDbApi rocksApi;
+    private final RocksDbSnapshotGenerator<DiskBackedCorfuTable<K, V>> rocksDbSnapshotGenerator;
+    private final ColumnFamilyRegistry columnFamilyRegistry;
+    private final ISerializer serializer;
+    // Index.
+    private final Map<String, String> secondaryIndexesAliasToPath;
+    private final Map<String, Byte> indexToId;
+    private final Set<Index.Spec<K, V, ?>> indexSpec;
+    // Metrics.
+    private final String metricsId;
+    // Options.
+    private final PersistenceOptions persistenceOptions;
+    @Getter
+    private final Statistics statistics;
+
+    public DiskBackedCorfuTable(@NonNull PersistenceOptions persistenceOptions,
+                                @NonNull Options rocksDbOptions,
+                                @NonNull ISerializer serializer,
+                                @Nonnull Index.Registry<K, V> indices) {
+
+        this.persistenceOptions = persistenceOptions;
+        this.secondaryIndexesAliasToPath = new HashMap<>();
+        this.indexToId = new HashMap<>();
+        this.indexSpec = new HashSet<>();
+
+        byte indexId = 0;
+        for (Index.Spec<K, V, ?> index : indices) {
+            this.secondaryIndexesAliasToPath.put(index.getAlias().get(), index.getName().get());
+            this.indexSpec.add(index);
+            this.indexToId.put(index.getName().get(), indexId++);
+        }
+
+        try {
+            this.statistics = new Statistics();
+            this.statistics.setStatsLevel(StatsLevel.ALL);
+            rocksDbOptions.setStatistics(statistics);
+
+            final RocksDbStore<DiskBackedCorfuTable<K, V>> rocksDbStore = new RocksDbStore<>(
```

Comment:
I think it might be outside the goals of this PR but just to make a note here - one RocksDB per one disk-backed table will introduce thread overhead per table.

Reply:
You bring up a very good point. There is a way to mitigate this, but I did not want to introduce a premature optimizations at this point. Basically, you can do:

      final Env sharedEnv = Env.getDefault()
                                .setBackgroundThreads(4)
                                .setHighPriorityBackgroundThreads(2);

      final Options options = new Options()
                              .setCreateIfMissing(true)
                              .setEnv(sharedEnv);